common/db/openssl/openssl.go

```diff
@@ -15,9 +16,10 @@ import (
 
 // For connecting to the database over ssl
 type SSLDBConnector struct {
-	dialInfo  *mgo.DialInfo
-	dialError error
-	ctx       *openssl.Ctx
+	dialInfo       *mgo.DialInfo
+	dialErrorMutex sync.Mutex
+	dialError      error
```

Comment:
after speaking with @gabrielrussell, it seems that dialError is actually somewhat broken as is. It is shared memory in a function that can be called concurrently. Putting a mutex stops it from being set simultaneously, but doesn't stop it from then being immediately overwritten before being read in  GetNewSession. Because of this, it should be removed for now.